src/binder/bindertracing.cpp

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ============================================================
+//
+// bindertracing.cpp
+//
+
+
+//
+// Implements helpers for binder tracing
+//
+// ============================================================
+
+#include "common.h"
+#include "bindertracing.h"
+
+#include "activitytracker.h"
+
+#ifdef FEATURE_EVENT_TRACE
+#include "eventtracebase.h"
+#endif // FEATURE_EVENT_TRACE
+
+using namespace BINDER_SPACE;
+
+namespace
+{
+    const WCHAR *s_activityName = W("AssemblyBind");
+
+    thread_local bool s_trackingBind = false;
```

Comment:
Is this thread local going to be used for other stuff? It is useless right now.

Reply:
The intention is that we will use it when determining if we care about other (non-start/stop) binder events for the current bind that is happening.

Reply:
Is it just so the trace always has a complete set of Start/.../Stop events? 

If somebody turns on the tracing in the middle of the bind, they may get Stop event without a matching Start event and that's by design.

Reply:
I had it more for ignoring the SPCL resources bind events - at least for this PR, the entire set of Start/.../Stop would not be fired.

Yeah, I'd think it makes sense to get the events from that point if tracing is turned on in the middle - the caveat being that the correlation would be missing, since there was no activity started. (Although this change doesn't do that - was going to update to that when we actually populate the event properties)

Reply:
As for this variable itself, I can remove it from this. As you said, it is useless right now and the next change I'm planning to make wouldn't actually use it either for what I had in mind (and maybe we'd do something different when we get there).